subsys/net/lib/socks/socks.c

```diff
@@ -80,29 +96,30 @@ int socks5_client_tcp_connect(const struct sockaddr *proxy,
 	/* size + 1 because just one method is supported */
 	size = sizeof(struct socks5_method_request_common) + 1;
 
-	ret = socks5_tcp_send(fd, (u8_t *)&mthd_req, size);
+	ret = net_context_sendto(ctx, (u8_t *)&mthd_req, size,
+				 proxy, proxy_len, NULL, K_NO_WAIT,
+				 ctx->user_data);
 	if (ret < 0) {
-		(void)close(fd);
 		LOG_ERR("Could not send negotiation packet");
-		return ret;
+		errno = ret;
```

Comment:
It should return negative error code, `errno` will be overwritten by the socket layer anyway.